vignettes/flowmaps-static.qmd

```diff
@@ -0,0 +1,774 @@
+   id_origin id_destination      n_trips
+   <chr>     <chr>                 <dbl>
+ 1 Andalusia Andalusia         21991628.
+ 2 Andalusia Aragon                1405.
+ 3 Andalusia Asturias               777.
+ 4 Andalusia Balearic Islands      1568.
+ 5 Andalusia Basque Country         970.
+ 6 Andalusia Canary Islands        1853.
+ 7 Andalusia Cantabria              268.
+ 8 Andalusia Castile and León      3025.
+ 9 Andalusia Castile-La Mancha    11374.
+10 Andalusia Catalonia             5679.
+# ℹ 339 more rows
+# ℹ Use `print(n = ...)` to see more rows
+```
+
+## Reshaping the data for visualization
+
+We are going to use the `{flowmapper}` [@flowmapper-r] package to plot the flows. This package expects the data to be in the following format:
+
+```
+'data.frame':	7 obs. of  8 variables:
+ $ id_a   : chr  "X1" "X2" "X3" "X3" ...
+ $ id_b   : chr  "X5" "X4" "X1" "X5" ...
+ $ xa     : num  2 14 10 10 2 14 14
+ $ ya     : num  6 10 9 9 6 10 10
+ $ xb     : num  10 4 2 10 4 10 10
+ $ yb     : num  4 10 6 4 10 4 9
+ $ flow_ab: num  1 2 3 3 1 1 4
+ $ flow_ba: num  2 3 2 5 2 1 5
+```
+
+
+### Reshape the flows
+
+The data we have right now in `flows_by_ca` is in a classic long format. That is, we have rows for both the number of trips from A to B and from B to A. `{flowmapper}` requires the data to be in different format, where there is only one row for each pair of `id_a` and `id_b` and two columns, one for the flow from `id_a` to `id_b` and one for the flow from `id_b` to `id_a`.
+
+```{r}
+flows_by_ca_twoway <- flows_by_ca |>
+  # Rename columns to match the desired output
+  rename(id_a = id_origin, id_b = id_destination, flow_ab = n_trips) |>
+  # Create the reverse flow
+  full_join(
+    flows_by_ca |>
+      rename(id_b = id_origin, id_a = id_destination, flow_ba = n_trips),
+    by = c("id_a", "id_b")
+  ) |>
+  # Replace NA with 0 for flows that don't exist in one direction
+  mutate(
+    flow_ab = coalesce(flow_ab, 0),
+    flow_ba = coalesce(flow_ba, 0)
+  ) |>
+  # Remove duplicate rows (where id_a and id_b are swapped)
+  filter(id_a <= id_b) |>
+  # Sort by id_a and id_b
+  arrange(id_a, id_b) |> 
+  # Remove the flows within the autonomous communities
+  filter(id_a != id_b)
+
+head(flows_by_ca_twoway)
+```
+
+```
+# A tibble: 6 × 4
+  id_a      id_b             flow_ab flow_ba
+  <chr>     <chr>              <dbl>   <dbl>
+1 Andalusia Aragon             1405.   1166.
+2 Andalusia Asturias            777.    596.
+3 Andalusia Balearic Islands   1568.   1362.
+4 Andalusia Basque Country      970.   1016.
+5 Andalusia Canary Islands     1853.   1494.
+6 Andalusia Cantabria           268.    167.
+```
+
+### Add coordinates
+
+We need the coordinates for each origin and destination. We can use the centroids of `spain_for_vis` polygons for that. Even though it has incorrect coordinates for the Canary Islands, this is actually what we want, as we will be plotting the flows on a map of Spain stored in the `spain_for_vis` object.
```

Comment:
same as above, can `{od}` do this with one function instead of my code block below?